File: src/main/java/com/negosud/api/service/FamilleService.java
package com.negosud.api.service;

import com.negosud.api.dto.request.FamilleRequestDto;
import com.negosud.api.dto.response.FamilleResponseDto;

import java.util.List;

public interface FamilleService {

    List<FamilleResponseDto> findAll();

    FamilleResponseDto findById(Long id);

    FamilleResponseDto create(FamilleRequestDto dto);

    FamilleResponseDto update(Long id, FamilleRequestDto dto);

    void delete(Long id);
}

File: front/src/services/familleService.js
import api from './api';

const BASE = '/familles';

export const familleService = {
  findAll: () => api.get(BASE),
  findById: (id) => api.get(`${BASE}/${id}`),
  create: (data) => api.post(BASE, data),
  update: (id, data) => api.put(`${BASE}/${id}`, data),
  delete: (id) => api.delete(`${BASE}/${id}`),
};
